credit-risk/src/app.py

```diff
@@ -1,28 +1,234 @@
+APPROVAL_THRESHOLD = 0.35
 
 
 def init(q: Q):
-    q.app.customer_status = {}
+    # build model
+    q.app.model = Model(TRAIN_CSV, ID_COLUMN, TARGET_COLUMN)
+    # prep df for analyzation
+    df = pd.read_csv(TEST_CSV).head(20)
+    df.drop(TARGET_COLUMN, axis=1, inplace=True)
+    # analyze df
+    q.app.predictions_df = q.app.model.predict(df)
+    q.app.contributions_df = q.app.model.contrib(df)
+    # prep df for viewing
+    df['Default Prediction Rate'] = q.app.predictions_df.round(4)
+    df.insert(loc=0, column='Status', value='Pending', allow_duplicates=True) 
+    q.app.customer_df = df
+
+
+def layout(q: Q):
+    q.page['meta'] = ui.meta_card(
+        box='', 
+        title='Credit Risk', 
+        layouts=[
+            ui.layout(
+                breakpoint='xs',
+                zones=[
+                    ui.zone('header'),
+                    ui.zone('customer_table'),
+                    ui.zone(
+                        'customer_page',
+                        zones=[
+                            ui.zone('customer_risk_explanation'),
+                            ui.zone('customer_shap_plot', size='600px'),
+                            ui.zone('button_group'),
+                            ui.zone('customer_features'),
+                        ]
+                    )
+                ]
+            ),
+            ui.layout(
+                breakpoint='m',
+                zones=[
+                    ui.zone('header'),
+                    ui.zone('customer_table'),
+                    ui.zone(
+                        'customer_page',
+                        direction=ui.ZoneDirection.ROW,
+                        zones=[
+                            ui.zone(
+                                'content',
+                                zones=[
+                                    ui.zone('customer_risk_explanation'),
+                                    ui.zone('customer_shap_plot', size='600px'),
+                                    ui.zone('button_group'),
+                                ]
+                            ),
+                            ui.zone('customer_features', size='300px'),
+                        ]
+                    )
+                ]
+            )
+        ]
+    )
+    q.page['header'] = ui.header_card(
+        box='header',
+        title='Credit Card Risk',
+        subtitle='Review customer ability to pay credit card bills',
+        icon='PaymentCard',
+        nav=[
+            ui.nav_group(
+                'Navigation', 
+                items=[
+                    ui.nav_item(name='render_customer_selector', label='Customers'),
+                ]
+            ),
+            ui.nav_group(
+                'Options', 
+                items=[
+                    ui.nav_item(name='dark_mode', label='Dark Mode'),
+                    ui.nav_item(name='light_mode', label='Light Mode'),
+                ]
+            )
+        ]
+    )
+
+
+def handle_approve_click(q: Q):
```

Comment:
Consider using `handle_on` and `on` for these button clicks

Reply:
Done